lib/Myriad/Service/Remote.pm

```diff
@@ -23,12 +23,17 @@ use Myriad::Service::Storage::Remote;
 
 has $myriad;
 has $service_name;
+has $local_service_name;
 has $storage;
 
 BUILD(%args) {
     weaken($myriad = delete $args{myriad});
     $service_name = delete $args{service_name} // die 'need a service name';
-    $storage = Myriad::Service::Storage::Remote->new(prefix => $service_name, storage => $myriad->storage);
+    $local_service_name = delete $args{local_service_name} // die 'need a local service name';
+    $storage = Myriad::Service::Storage::Remote->new(
+                                                prefix => $service_name,
+                                                storage => $myriad->storage,
+                                                local_service_name => $local_service_name);
```

Comment:
this one here too https://github.com/binary-com/perl-Myriad/pull/76/files#diff-60023cc37603429cad04d7246dc1a75d48098b6d064f3f459683874e63651ddcR53

im thinking instead of having to deal with creating myriad objects and components, by passing different information as parameters can get confusing. and this can lead to difficulties in following logic pretty soon.
Why don't we make it a standard that all components will take `myriad => $myriad, service => $service` 
this way we can encapsulate all information within these two objects only without the need of searching, where can I get this info from.  Also will provide us better organising strategy.

Reply:
I see the point  here, however:

`$service_name` is from the developer code.
`$storage` is a Myriad object indeed.
`$local_service_name` is needed for tracing/metrics

three different information from three different sources, I don't think that `myriad` and `service` are enough

Reply:
but in general the flow of how we pass things around especially `$myriad` needs a review

Reply:
`$service_name`  is already part of `Service` object https://github.com/binary-com/perl-Myriad/blob/master/lib/Myriad/Service/Implementation.pm#L96 and `$local_service_name` can be part of it too.
so if we passed only `$service` & `$myriad` as a standard way should be enough, as those two objects carries all needed information